lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -2923,4 +2923,23 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
             /*layout=*/cstNone, /*requires_grad=*/cstFalse);
         return success();
       });
+  patterns.onOp("NonMaxSuppression", 10,
+                [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+                  Torch::ValueTensorType resultType;
+                  SmallVector<Value> operands;
+                  if (binder.tensorOperandsList(operands) ||
+                      binder.tensorResultType(resultType))
+                    return failure();
+
+                  // TODO: Add support for handling max_output_boxes_per_class
+                  // and score_threshold arg.
```

Comment:
Ah, they are optional inputs. These actually don't get added as operands if they are left unspecified. 

e.g., with all of the optional arguments for this op left blank, here's a sample:

```MLIR
module {
  func.func @nms_graph(%arg0: !torch.vtensor<[1,6,4],f32>, %arg1: !torch.vtensor<[1,1,6],f32>) -> !torch.vtensor<[3,3],si64> attributes {torch.onnx_meta.ir_version = 10 : si64, torch.onnx_meta.opset_version = 19 : si64, torch.onnx_meta.producer_name = "", torch.onnx_meta.producer_version = ""} {
    %none = torch.constant.none
    %0 = torch.operator "onnx.NonMaxSuppression"(%arg0, %arg1) : (!torch.vtensor<[1,6,4],f32>, !torch.vtensor<[1,1,6],f32>) -> !torch.vtensor<[3,3],si64> 
    return %0 : !torch.vtensor<[3,3],si64>
  }
}
```

Reply:
This is actually somewhat annoying for this op, since hoping that `iouThreshold = operands[3]` is not reliable (it might even cause a crash when used on the example above).

Reply:
I'd say that we need a few match failures:

1. if there are three operands, and the last operand has a float dtype, we cannot infer which argument it is.
2. if there are four operands, and there is an int and a float optional, then we cannot infer which argument is missing.
3. if center_point_box is not zero (default) we should indicate a match failure since this is not handled. 

I'm not sure what to do about the other optional arguments. This is a rather confusing onnx op.